fingerprint2.js

```diff
@@ -591,6 +612,97 @@
       document.body.appendChild(ads);
       return document.getElementById("ads") ? false : true;
     },
+    getHasLiedLanguages: function(){
+      //We check if navigator.language is equal to the first language of navigator.languages
+      if(typeof navigator.languages !== "undefined"){
+        try{
+          var firstLanguages = navigator.languages[0].substr(0, 2);
+          if(firstLanguages !== navigator.language.substr(0, 2)){
+            return true;
+          }
+        }catch(err){
+          return true;
+        }
+      }
+
+      return false;
+    },
+    getHasLiedResolution: function(){
+      if(screen.width < screen.availWidth){
+        return true;
+      }
+      if(screen.height < screen.availHeight){
+        return true;
+      }
+
+      return false;
+    },
+    getHasLiedOs: function(){
+      var userAgent = navigator.userAgent;
+      var oscpu = navigator.oscpu;
+      var platform = navigator.platform;
+      var os;
+      //We extract the OS from the user agent (respect theorder of the if else if statement)
+      if(userAgent.toLowerCase().indexOf("windows phone") >= 0){
+        os = "Windows Phone";
+      } else if(userAgent.toLowerCase().indexOf("win") >= 0){
+        os = "Windows";
+      } else if(userAgent.toLowerCase().indexOf("android") >= 0){
+        os = "Android";
+      } else if(userAgent.toLowerCase().indexOf("linux") >= 0){
+        os = "Linux";
+      } else if(userAgent.toLowerCase().indexOf("iPhone") >= 0 || userAgent.toLowerCase().indexOf("iPad") >= 0 ){
+        os = "iOS";
+      } else if(userAgent.toLowerCase().indexOf("mac") >= 0){
+        os = "Mac";
+      } else{
+        os = "Other";
+      }
+      //We detect if the person uses a mobile device
+      var mobileDevice;
+      if (("ontouchstart" in window) ||
+           (navigator.maxTouchPoints > 0) ||
+           (navigator.msMaxTouchPoints > 0)) {
+            mobileDevice = true;
+      } else{
+        mobileDevice = false;
+      }
+
+      if(mobileDevice && os !== "Windows Phone" && os !== "Android" && os !== "iOS" && os !== "Other"){
+        return true;
+      }
+
+      //We compare oscpu with the os extracted from the ua
+      if(typeof oscpu !== "undefined"){
+        if(oscpu.toLowerCase().indexOf("win") >= 0 && os !== "Windows" && os !== "Windows Phone"){
+          return true;
+        }else if(oscpu.toLowerCase().indexOf("linux") >= 0 && os !== "Linux" && os !== "Android"){
```

Comment:
please update spacing here too. Sorry for this, I will add an eslint rule to verify this automatically.